lib/kcerts/ssh.go

```diff
@@ -103,6 +107,52 @@ func (a SSHAgent) Valid() bool {
 	return err == nil
 }
 
+func (a *SSHAgent) GetStandardSocketPath() (string, error) {
+	path, err := GetConfigDir("enkit")
+	if err != nil {
+		return "", err
+	}
+
+	// Securely make sure path exists and its permissions are 0700
+	if err := os.MkdirAll(path, 0700); err != nil && !os.IsExist(err) {
+		return "", err
+	}
+
+	socket := filepath.Join(path, "agent")
+	return socket, nil
+}
+
+func (a *SSHAgent) UseStandardPaths() error {
+	socket, err := a.GetStandardSocketPath()
+	if err != nil {
+		return err
+	}
+
+	if a.Socket == socket {
+		// no standardization needed.
+		return nil
+	}
+
+	// Create symlink with a random name
+	path, err := GetConfigDir("enkit")
+	if err != nil {
+		return err
+	}
+	tempname := fmt.Sprintf("%s/enkit.tmp%016x", path, mathrand.New(srand.Source).Uint64())
+	defer os.Remove(tempname)
+	if err := os.Symlink(a.Socket, tempname); err != nil {
+		return err
+	}
+	// Rename symlink to the standard name
+	os.Remove(socket) // if it exists
```

Comment:
nit: `os.Remove` can go? `os.Rename` behaves like the syscall, "Rename renames (moves) oldpath to newpath. If newpath already exists and is not a directory, Rename replaces it." https://pkg.go.dev/os#Rename

(that's how it avoids race conditions - this is small and minor here, so not a big deal, but we should remove it)